packages/itwinui-react/src/core/Table/TableExpandableRow.tsx

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import * as React from 'react';
+import type { Row } from '../../react-table/react-table.js';
+import { Box, useMergedRefs } from '../../utils/index.js';
+import type { Virtualizer, VirtualItem } from '@tanstack/react-virtual';
+
+export const TableExpandableRow = <T extends Record<string, unknown>>(props: {
```

Comment:
We should `memo`ize this component too.

Also I think some of the props can be removed/reorganized. For example `subComponent` could just be passed through `children`, and `tableRowRef` and `virtualizer` could be handled using a normal `forwardRef` pattern.

Reply:
I'm not quite sure how to combine the 2 refs before passing into the child component as forwarded ref so I passed the tableRef then merged them together inside the child. Is this the correct way to do this?

I have memoized the component.